be/src/exec/pipeline/runtime_filter_types.h

```diff
@@ -0,0 +1,336 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#pragma once
+#include <memory>
+#include <mutex>
+
+#include "common/statusor.h"
+#include "exprs/expr_context.h"
+#include "exprs/predicate.h"
+#include "exprs/vectorized/runtime_filter_bank.h"
+
+namespace starrocks {
+namespace pipeline {
+class RuntimeFilterHolder;
+using RuntimeFilterHolderPtr = std::unique_ptr<RuntimeFilterHolder>;
+using RuntimeInFilter = starrocks::ExprContext;
+using RuntimeBloomFilter = starrocks::vectorized::RuntimeFilterBuildDescriptor;
+using RuntimeBloomFilterProbeDescriptor = starrocks::vectorized::RuntimeFilterProbeDescriptor;
+using RuntimeBloomFilterProbeDescriptorPtr = RuntimeBloomFilterProbeDescriptor*;
+using RuntimeBloomFilterRunningContext = starrocks::vectorized::JoinRuntimeFilter::RunningContext;
+using RuntimeInFilterPtr = RuntimeInFilter*;
+using RuntimeBloomFilterPtr = RuntimeBloomFilter*;
+using RuntimeInFilters = std::list<RuntimeInFilterPtr>;
+using RuntimeBloomFilters = std::list<RuntimeBloomFilterPtr>;
+struct RuntimeFilterCollector;
+using RuntimeFilterCollectorPtr = std::unique_ptr<RuntimeFilterCollector>;
+using RuntimeFilterProbeCollector = starrocks::vectorized::RuntimeFilterProbeCollector;
+using Predicate = starrocks::Predicate;
+struct RuntimeBloomFilterBuildParam;
+using RuntimeBloomFilterBuildParams = std::list<RuntimeBloomFilterBuildParam>;
+// Parameters used to build runtime bloom-filters.
+struct RuntimeBloomFilterBuildParam {
+    RuntimeBloomFilterBuildParam(bool eq_null, const ColumnPtr& column, size_t column_offset, size_t ht_row_count)
+            : eq_null(eq_null), column(column), column_offset(column_offset), ht_row_count(ht_row_count) {}
+    bool eq_null;
+    ColumnPtr column;
+    size_t column_offset;
+    size_t ht_row_count;
+};
+
+// RuntimeFilterCollector contains runtime in-filters and bloom-filters, it is stored in RuntimeFilerHub
+// and every HashJoinBuildOperatorFactory has its corresponding RuntimeFilterCollector.
+struct RuntimeFilterCollector {
```

Comment:
The concepts for RuntimeFilter are very confusing.

Reply:
RuntimeFilterCollector contains both RuntimeInFilters and RuntimeBloomFilters. another good name is acceptable.